redux/actions/currenciesActions.ts

```diff
@@ -20,31 +20,36 @@ export const getUserCurrenciesFromAsync = () => async (dispatch: Dispatch<IGetCu
   }
 }
 
-export const addCurrencyBySymbol = (valueSearched: string) => async (dispatch: Dispatch, getState: () => IState) => {
-  const {allCurrencies, userCurrencyList} = getState().currencies
-
-  const newCrypto: ICurrency | undefined  = allCurrencies.find( ({symbol, name}) => {
-    let isAdded: ICurrency | undefined  = userCurrencyList.find( ({symbol, name}) => (
-      symbol.toLowerCase() === valueSearched.toLowerCase() ||  name.toLowerCase() === valueSearched.toLowerCase()
-    ))
-
-    return (symbol.toLowerCase() === valueSearched.toLowerCase() ||  name.toLowerCase() === valueSearched.toLowerCase()) && !isAdded
-  })
+export const addCurrencyBySymbol = (valueSearched: string) => async (dispatch: Dispatch<IGetCurrenciesAction>, getState: () => IState) => {
+  const {userCurrencyList} = getState().currencies
 
-  if (newCrypto) {
-    const updatedUserCurrencies: ICurrency[] = [...userCurrencyList, newCrypto]
+  const host = 'https://data.messari.io/api/v1/assets'
+  const fields = 'metrics?fields=id,symbol,name,market_data/price_usd,market_data/percent_change_usd_last_24_hours'
 
-    dispatch({type: ADD_NEW_CRYPTOCURRENCY, payload: updatedUserCurrencies})
-
-    try {
-      const updatedList: string = JSON.stringify(updatedUserCurrencies)
-      await AsyncStorage.setItem(AsyncKeys.userCurrencyList, updatedList)
-    } catch (e) {
-      console.log(e)
+  try {
+    const response = await axios.get(`${host}/${valueSearched.toLocaleLowerCase()}/${fields}`)
+    
+    if(response.data.data) {
+      const isAdded: ICurrency | undefined  = userCurrencyList.find( ({symbol, name}) => (
+        symbol.toLowerCase() === valueSearched.toLowerCase() ||  name.toLowerCase() === valueSearched.toLowerCase()
+      ))
+
+      if (!isAdded) {
+        const updatedList = [...userCurrencyList, response.data.data]
+        dispatch({ type: ADD_NEW_CRYPTOCURRENCY, payload: updatedList }) 
+
+        try {
+          await AsyncStorage.setItem(AsyncKeys.userCurrencyList, JSON.stringify(updatedList))
+        } catch (e) {
```

Comment:
[question] Does the backend returns a certain error response? Generally it follows this structure: `response.error`